lib/internal/process/pre_execution.js

```diff
@@ -545,15 +545,20 @@ function initializeDeprecations() {
     });
   }
 
-  if (pendingDeprecation) {
+  if (getOptionValue('--deprecated-process-binding')) {
     process.binding = deprecate(process.binding,
                                 'process.binding() is deprecated. ' +
                                 'Please use public APIs instead.', 'DEP0111');
-
-    process._tickCallback = deprecate(process._tickCallback,
-                                      'process._tickCallback() is deprecated',
-                                      'DEP0134');
+  } else {
+    // Ideally it wouldn't be defined at all already but realm.js does not
+    // provide a means of checking to see if the option is enabled or not,
+    // so we do it here.
+    process.binding = undefined;
```

Comment:
Why not `delete process.binding` ?

Reply:
Simply because `delete` generally leads to perf issues so I tend to avoid using it, but either works. Ultimately what I'd rather do is not set `process.binding` at all in realm.js when the option is set but that's currently not possible.

Reply:
`delete` is closer to not setting it. I'd prefer to not see the property it all when it's disabled.

Reply:
fair enough

Reply:
done!